lib/scrabble/scrabble.rb

```diff
@@ -0,0 +1,92 @@
+module Scrabble
+  class Scrabble
+    VALUES = {a: 1, e: 1, i: 1, o: 1, u: 1, l: 1, n: 1, r: 1, s: 1, t: 1,
+      d: 2, g: 2,
+      b: 3, c: 3, m: 3, p: 3,
+      f: 4, h: 4, v: 4, w: 4, y: 4,
+      k: 5,
+      j: 8, x: 8,
+      q: 10, z: 10
+    }
+
+    # .score(word): returns the total score value for the given word [breakfast]
+    def self.score(word) # word is input as a string (case insensitive)
+      letters = convert_input(word)
+      letters.inject(0){ |sum, letter| sum + VALUES[letter.to_sym] }
+    end
+
+    # returns the word as an array of lowercase letters, if input is acceptable
+    def self.convert_input(word)
+      acceptable_input_check(word)
+      word.downcase.chars
+    end
+
+    # checks word for not-allowed content. If contains, prompts for new input until accepted.
+    def self.acceptable_input_check(word)
+      # checks for nil, special characters. NOTE: '' is treated as zero, and is an accepted input.
+      status = (
+      if word.is_a?(String) # checks for nil input, arrays, integers (not in a string), etc.
+        word.chars.each do |letter|
+          return false unless ('a'..'z').include?(letter) || ('A'..'Z').include?(letter) #.downcase doesn't work on integers
+        end
+        word.length <= 7
```

Comment:
word.length <= 7? Scrabble doesn't allow for words longer than 7 characters, so this was  to prevent 8+ characters.